### ISSUE TITLE
A Follow-up Requirement Should be Added - 1863. Sum of All Subset XOR Totals

### ISSUE THREAD

**Issue by CarlCzerny837** [LeetCode-Feedback/LeetCode-Feedback#5313]
Your LeetCode username
v5qYY4Q65w

Category of the bug
- [ ] Question
- [ ] Solution
- [ ] Language
- [√] Missing Test Cases 

Description of the bug
The solution with O(2^n) time complexity can be accepted now. But there is an solution with O(n) time complexity, and the solution with brute force for this problem is too easy for a real interview. So, I think a followed-up requirement should be added, and user should try to find a solution with O(n) time complexity.

Code you used for Submit/Run operation

The time complexity of the solution below is O(2^n). 
```
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        res = 0
        n = len(nums)
        def dfs(val, idx):
            nonlocal res
            if idx == n:
                res += val
                return
            dfs(val ^ nums[idx], idx + 1)
            dfs(val, idx + 1)
        
        dfs(0, 0)
        return res
```

The time complexity of the solution below is O(n).
```
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        res = 0
        n = len(nums)
        for num in nums:
            res |= num
        return res << (n - 1)
```

Language used for code
Python3

Expected behavior
The requirement below should be added:
**Follow-up**: can you write an algorithm in O(n) time complexity?

**Comment by Sky-Nik** [comment]
You may be familiar with the concept of Weekly and Biweekly contests. In such contests, Q1 is usually a problem with exceptionally low constraints, on purpose. 

Besides, one may not simply add a testcase with a large N. It would require at least modding the answer as `res << (n - 1)` is a clear overflow in C++/Java. This would also render many current discuss solutions wrong -- something that users hate seeing.

To sum up, I doubt that such a change of constraints will be introduced. I would instead consider adding it in Notes as a Challenge of sorts.

**Comment by CarlCzerny837** [comment]
> You may be familiar with the concept of Weekly and Biweekly contests. In such contests, Q1 is usually a problem with exceptionally low constraints, on purpose.
> 
> Besides, one may not simply add a testcase with a large N. It would require at least modding the answer as res << (n - 1) is a clear overflow in C++/Java. This would also render many current discuss solutions wrong -- something that users hate seeing.
> 
> To sum up, I doubt that such a change of constraints will be introduced. I would instead consider adding it in Notes as a Challenge of sorts.

I agree with you , now I change the expected behavior to "a follow-up requirement should be added". 
I am familiar with the weekly and biweekly contests, but I am not familiar with bit manipulation. The nums.length can not be increased to 10^5, and writing an algorithm with O(n) time complexity can be a follow-up task.

**Comment by LC-Sue** [comment]
Hi @CarlCzerny837 
Thank you for reaching out to us. I've relayed this issue to our team to investigate.

**Comment by LC-Sue** [comment]
Hi @CarlCzerny837
Thank you for your time. The problem was set to intentionally allow the exponential solution. Our team has decided not to add the follow up as it has a very low probability to appear in an interview and it is more related to competitive programming.
